test/device/link_to_device_test.cpp

```diff
@@ -0,0 +1,41 @@
+#include <gtest/gtest.h>
+
+#include "utils.hpp"
+
+namespace test {
+
+class LinkToDevice : public testing::Test {
+public:
+    void TearDown() override {};
+    void SetUp() override {};
+};
+
+TEST_F(LinkToDevice, NoLinkToDevice) {
+    auto source = std::make_shared<sim::RoutingModule>(sim::RoutingModule());
+    auto dest = std::make_shared<sim::RoutingModule>(sim::RoutingModule());
+
+    EXPECT_EQ(source->get_link_to_destination(dest), nullptr);
+}
```

Comment:
This test is pretty strange. When `source` creates, it does not know nothing about `dest`, so `EXPECT_EQ` always passes. If you want to test that `source->get_link_to_destination` might return false, you may do add inlink to dest, but do not update routing table of source (also it shows that these actions are independent):
```suggestion
TEST_F(LinkToDevice, NoLinkToDevice) {
    auto source = std::make_shared<sim::RoutingModule>(sim::RoutingModule());
    auto dest = std::make_shared<sim::RoutingModule>(sim::RoutingModule());
    auto link = std::shared_ptr<TestLink>(source, dest);
    dest.add_inlink(source);

    EXPECT_EQ(source->get_link_to_destination(dest), nullptr);
}
```